linux/Platform.c

```diff
@@ -347,32 +344,28 @@ void Platform_getDiskIO(unsigned long int *bytesRead, unsigned long int *bytesWr
    *bytesRead = 512 * read_sum;
    *bytesWrite = 512 * write_sum;
    *msTimeSpend = timeSpend_sum;
+   return true;
 }
 
-void Platform_getNetworkIO(unsigned long int *bytesReceived,
+bool Platform_getNetworkIO(unsigned long int *bytesReceived,
                            unsigned long int *packetsReceived,
                            unsigned long int *bytesTransmitted,
                            unsigned long int *packetsTransmitted) {
    FILE *fd = fopen(PROCDIR "/net/dev", "r");
-   if (!fd) {
-      *bytesReceived = 0;
-      *packetsReceived = 0;
-      *bytesTransmitted = 0;
-      *packetsTransmitted = 0;
-      return;
-   }
+   if (!fd)
+      return false;
 
    unsigned long int bytesReceivedSum = 0, packetsReceivedSum = 0, bytesTransmittedSum = 0, packetsTransmittedSum = 0;
    char lineBuffer[512];
    while (fgets(lineBuffer, sizeof(lineBuffer), fd)) {
       char interfaceName[32];
       unsigned long int bytesReceivedParsed, packetsReceivedParsed, bytesTransmittedParsed, packetsTransmittedParsed;
-      if (fscanf(fd, "%31s %lu %lu %*u %*u %*u %*u %*u %*u %lu %lu %*u %*u %*u %*u %*u %*u",
-                     interfaceName,
-                     &bytesReceivedParsed,
-                     &packetsReceivedParsed,
-                     &bytesTransmittedParsed,
-                     &packetsTransmittedParsed) != 5)
+      if (sscanf(lineBuffer, "%31s %lu %lu %*u %*u %*u %*u %*u %*u %lu %lu %*u %*u %*u %*u %*u %*u",
```

Comment:
Good catch with the fscanf vs sscanf there.  I think you can safely drop the trailing "%*u %*u %*u %*u %*u %*u" in the above string - that's going to scan those unsigned integers for nothing and throw the result away, so it should be OK to stop scanning at the final %lu there.

Reply:
yep; dropped trailing ignored format specifiers